nilearn/regions/region_extractor.py

```diff
@@ -0,0 +1,337 @@
+        is set with this string as a folder name in the directory.
+        passed to NiftiMapsMasker.
+
+    memory_level: int, default 0, optional
+        Aggressiveness of memory catching. The higher the number, the higher
+        the number of functions that will be cached. Zero mean no caching.
+        passed to NiftiMapsMasker.
+
+    verbose: int, default 0, optional
+        Indicates the level of verbosity by printing the message. Zero indicates
+        nothing is printed.
+
+    Attributes
+    ----------
+    regions_img_: Nifti1Image object
+        list of separated regions with each region lying on a 3D volume
+        concatenated into a 4D image.
+
+    index_: numpy array
+        array of list of indices where each index value is assigned to
+        each separate region of its corresponding family of brain maps.
+
+    References
+    ----------
+    * Abraham et al. "Region segmentation for sparse decompositions: better
+      brain parcellations from rest fMRI", Sparsity Techniques in Medical Imaging,
+      Sep 2014, Boston, United States. pp.8
+    """
+    def __init__(self, maps_img, mask_img=None, min_region_size=1350,
+                 threshold=1., thresholding_strategy='ratio_n_voxels',
+                 extractor='local_regions', standardize=False, detrend=False,
+                 low_pass=None, high_pass=None, t_r=None,
+                 memory=Memory(cachedir=None), memory_level=0, verbose=0):
+        super(RegionExtractor, self).__init__(
+            maps_img=maps_img, mask_img=mask_img,
+            standardize=standardize, detrend=detrend, low_pass=low_pass,
+            high_pass=high_pass, t_r=t_r, memory=memory,
+            memory_level=memory_level, verbose=verbose)
+        self.maps_img = maps_img
+        self.min_region_size = min_region_size
+        self.thresholding_strategy = thresholding_strategy
+        self.threshold = threshold
+        self.extractor = extractor
+
+    def fit(self, X=None, y=None):
+        """ Prepare the data and setup for the region extraction
+        """
+        maps_img = check_niimg_4d(self.maps_img)
+
+        list_of_strategies = ['ratio_n_voxels', 'img_value', 'percentile']
+        if self.thresholding_strategy not in list_of_strategies:
+            message = ("'thresholding_strategy' should be "
+                       "either of these {0}").format(list_of_strategies)
+            raise ValueError(message)
+
+        if self.threshold is None or isinstance(self.threshold, _basestring):
+            raise ValueError("The given input to threshold is not valid. "
+                             "Please submit a valid number specific to either of "
+                             "the strategy in {0}".format(list_of_strategies))
+        elif isinstance(self.threshold, numbers.Number):
+            # foreground extraction
+            if self.thresholding_strategy == 'ratio_n_voxels':
+                if not isinstance(self.threshold, float):
```

Comment:
My understanding is that self.treshold can be an int in 'ratio_n_voxels' case. In any case you should probably not do the parameter checking here since _threshold_maps_ratio already has some.